src/main/java/com/intuit/graphql/orchestrator/schema/transform/FieldResolverTransformerPreMerge.java

```diff
@@ -39,21 +42,51 @@ public XtextGraph transform(XtextGraph sourceXtextGraph) {
   }
 
   private void validateFieldWithResolver(List<FieldResolverContext> fieldResolverContexts) {
-    fieldResolverContexts
-        .forEach(fieldResolverContext -> {
-          FieldDefinition fieldDefinition = fieldResolverContext.getFieldDefinition();
-          String fieldName = fieldDefinition.getName();
-          String parentTypeName = XtextTypeUtils.getParentTypeName(fieldDefinition);
-
-          EObject parentType = fieldDefinition.eContainer();
-          if (!(canContainFieldResolverDirective(parentType))) {
-            throw new NotAValidLocationForFieldResolverDirective(fieldName, parentTypeName);
-          }
-
-          if (hasArguments(fieldDefinition)) {
-            throw new ArgumentDefinitionNotAllowed(fieldName, parentTypeName);
-          }
-        });
+      fieldResolverContexts
+              .forEach(fieldResolverContext -> {
+                  FieldDefinition fieldDefinition = fieldResolverContext.getFieldDefinition();
+                  String fieldName = fieldDefinition.getName();
+                  String parentTypeName = XtextTypeUtils.getParentTypeName(fieldDefinition);
+
+                  EObject parentType = fieldDefinition.eContainer();
+                  if (!(canContainFieldResolverDirective(parentType))) {
+                      throw new NotAValidLocationForFieldResolverDirective(fieldName, parentTypeName);
+                  }
+
+                  if (hasArguments(fieldDefinition)) {
+                      throw new ArgumentDefinitionNotAllowed(fieldName, parentTypeName);
+                  }
+
+                  EList<Directive> directives = fieldDefinition.getDirectives();
+                  boolean hasResolverDirective = false;
+                  boolean hasExternalDirective = false;
+                  boolean hasProvidesDirective = false;
+                  boolean hasRequiresDirective = false;
```

Comment:
https://github.com/graph-quilt/graphql-orchestrator-java/blob/master/src/main/java/com/intuit/graphql/orchestrator/utils/XtextUtils.java

See if this util will help get rid of switch case

Reply:
used this util